src/services/stringCompletions.ts

```diff
@@ -114,6 +114,14 @@ namespace ts.Completions.StringCompletions {
                 switch (parent.parent.kind) {
                     case SyntaxKind.TypeReference:
                         return { kind: StringLiteralCompletionKind.Types, types: getStringLiteralTypes(typeChecker.getTypeArgumentConstraint(parent as LiteralTypeNode)), isNewIdentifier: false };
+                    case SyntaxKind.ParenthesizedType: {
+                        const typeNode = findAncestor(parent.parent, n => !isParenthesizedTypeNode(n.parent)) as ParenthesizedTypeNode;
+                        const typeArguments = (typeNode.parent as NodeWithTypeArguments).typeArguments;

```

Comment:
What about

```ts
Foo[("hello")]
```

?

Reply:
I think in short, you should really consider wrapping this whole thing in a loop where you check the parents each time.

Reply:
@DanielRosenwasser Not sure that I got your idea :(. I thought about the issue like so, `type T3 = T2<(((((/**/)))))>;` parent is `ParenthesizedType`, we need to skip all `ParenthesizedType` to get correct type argument. Or am I wrong?

`Foo[("hello")]` it is not `ParenthesizedType`. this case will not be handled.